### ISSUE TITLE
Unable to use mutations with remote schema permissions

### ISSUE THREAD

**Issue by jflambert** [hasura/graphql-engine#8046]
### Version Information

Server Version: 2.1.1

### What is the expected behaviour?

Should be able to select mutations when setting remote schema permissions.

### What is the current behaviour?

Completely unable to select mutations in remote schema permissions. Whether combined with queries or standalone.

### How to reproduce the issue?

1. Enable remote schema permissions
2. Go to the web console, select a remote schema and the 
3. Expand mutation_root
4. Select any mutation
5. Click the save button at the bottom
6. Error popup in the top right

### Screenshots or Screencast

![image](https://user-images.githubusercontent.com/11539524/150046939-68f878d9-c603-4533-b138-6bdb6e8071c5.png)

**Comment by jflambert** [comment]
hasura error logs don't give any extra details

**Comment by abhi40308** [comment]
Hey @jflambert , currently the server doesn't allow to add a remote schema without a query root. The solution here would be to add a `query_root` field.
If no `Query` field is selected, the console does not send the `Query` object as shown in screenshots, resulting in the error.

<img width="629" alt="Screenshot 2022-01-19 at 9 43 53 PM" src="https://user-images.githubusercontent.com/26903230/150278534-bf182f9a-ebff-44b0-82b2-69624b1da2fe.png">

<img width="497" alt="Screenshot 2022-01-19 at 9 44 07 PM" src="https://user-images.githubusercontent.com/26903230/150278480-dff2e5ba-de05-43dc-9add-ff89b62e85c0.png">

**Comment by jflambert** [comment]
@abhi40308  this is not an option for me

I'm adding remote schemas that ONLY provide mutations.

**Comment by tirumaraiselvan** [comment]
@jflambert Are you adding another Hasura instance as a remote schema?

**Comment by jflambert** [comment]
> @jflambert Are you adding another Hasura instance as a remote schema?

No. It's a graphql remote schema powered by [graphql-dotnet](https://github.com/graphql-dotnet/graphql-dotnet).

**Comment by AesSedai** [comment]
I solved this by basically adding in a dummy query to my remote schema, ala
```
type Query {
  required: Int
}
```
and a simple resolver that always returned `1`, then enabling query permission for that in Hasura. As you mentioned, it may not be an option for you but it at least "ticked the box" that allowed it to work for me.

**Comment by jflambert** [comment]
@AesSedai yep I did exactly the same.

I still maintain Hasura's error message is completely unhelpful.
